sdk/resource/resource.go

```diff
@@ -17,30 +17,79 @@
 package resource
 
 import (
-	"reflect"
+	"sort"
+	"strings"
 
 	"go.opentelemetry.io/otel/api/core"
 )
 
 // Resource describes an entity about which identifying information and metadata is exposed.
 type Resource struct {
-	labels map[core.Key]core.Value
+	sorted []core.KeyValue
+	keys   map[core.Key]struct{}
 }
 
 // New creates a resource from a set of attributes.
 // If there are duplicates keys then the first value of the key is preserved.
 func New(kvs ...core.KeyValue) *Resource {
-	res := &Resource{
-		labels: map[core.Key]core.Value{},
-	}
+	res := &Resource{keys: make(map[core.Key]struct{})}
 	for _, kv := range kvs {
-		if _, ok := res.labels[kv.Key]; !ok {
-			res.labels[kv.Key] = kv.Value
+		// First key wins.
+		if _, ok := res.keys[kv.Key]; !ok {
```

Comment:
the reason why I prefer bools to struct{} for sets, minor preference, means you can just test `if res.keys[kv.Key] {` instead of needing the `ok`.

Reply:
Fair. I see it both ways. The slight size advantage, potential decrease in ambiguity around what the map is for, and the fact the compiler is going to deal with the blank identifier assignment as a no-op anyways has led me to default to the `map[T]struct{}` for. None of these are really strong reasons to choose one over the other and, overall, I don't have a strong preference 🤷‍♀.